src/main/java/frc/robot/subsystems/shooter/SetShooterVelocity.java

```diff
@@ -0,0 +1,92 @@
+package frc.robot.subsystems.shooter;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import java.util.function.DoubleSupplier;
+
+import static frc.robot.Robot.robotConstants;
+import static frc.robot.Robot.shooter;
+
+
+/**
+ * This command spins the wheel in the desired velocity in order to shoot the power cells.
+ */
+public class SetShooterVelocity extends CommandBase {
+
+    private static final int LOADED_CELLS_IN_AUTO = 3;
+    private DoubleSupplier velocitySetpoint;
+    private boolean isAuto;
+    private double lastSetpoint;
+    private boolean isInZone;
+    private double lastTimeOutsideZone;
+    private int cellsShot;
+
+    /**
+     * Constructs a shoot command with default RPM setpoint.
+     * @see frc.robot.constants.RobotConstants.ShooterConstants#DEFAULT_RPM
+     */
+    public SetShooterVelocity() {
+        this(robotConstants.shooterConstants.DEFAULT_RPM);
+    }
+    public SetShooterVelocity(boolean isAuto) {
+        this();
+        this.isAuto = isAuto;
+    }
+
+    /**
+     * @param velocitySetpoint The setpoint used for calculation the PID velocity error in RPM.
+     */
+    public SetShooterVelocity(double velocitySetpoint) {
+        this(() -> velocitySetpoint);
+    }
+
+    /**
+     * @param velocitySetpoint The setpoint used for calculation the PID velocity error in RPM.
+     * @param isAuto Whether the command is run in autonomous.
+     */
+    public SetShooterVelocity(double velocitySetpoint, boolean isAuto){
+        this(velocitySetpoint);
+        this.isAuto = isAuto;
+    }
+
+    public SetShooterVelocity(DoubleSupplier velocitySetpointSupplier) {
+        addRequirements(shooter);
+        velocitySetpoint = velocitySetpointSupplier;
+    }
+
+    @Override
+    public void initialize() {
+        lastSetpoint = velocitySetpoint.getAsDouble();
+        shooter.startPID(lastSetpoint);
+        cellsShot = 0;
+        isInZone = true;
+    }
```

Comment:
cellsShot = 0;
        isInZone = true;

initialize vars before using startPID
and add lastTimeOutsideZone initialize...?